src/Error.cpp

```diff
@@ -148,6 +149,34 @@ ErrorReport::ErrorReport(const char *file, int line, const char *condition_strin
     }
 }
 
+#ifdef HALIDE_WITH_EXCEPTIONS
+namespace {
+
+// This is a trick: rethrow the pending (unhandled) exception
+// so that we can see what it is and log `what` before dying.
+// Strictly speaking, standard C++ doesn't allow you to do a try/throw/catch
+// within a terminate handler, but in practice, Clang and GCC (at least)
+// allow it, so we'll use it to improve quality-of-life for our tests.
+void UnhandledExceptionHandler() {
+    try {
+        throw;
+    } catch (Error &e) {
```

Comment:
Neat, but might be better to do:
```
if (auto exn = std::current_exception()) {
    try {
        std::rethrow_exception(exn);
    } catch (Error &e) {
        // ...
    }
}

abort();
```
This way we don't fail to rethrow an exception, in case `terminate` was called for a reason besides failure-to-catch

Reply:
> Why do you say that? Consulting the [C++17 standard](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/n4659.pdf), sec 18.3p7-8, I read:

Interesting -- I had been under the impression that trying to do anything fancy in std::terminate was (technically) UB.

Reply:
> Neat, but might be better to do:
> 
> ```
> if (auto exn = std::current_exception()) {
> ```

Hmm... yeah, I'll give that a try.